### ISSUE TITLE
[scroll-animations] `ScrollTimeline` constructor has contradictory IDL and prose 

### ISSUE THREAD

**Issue by graouts** [w3c/csswg-drafts#11340]
The [`ScrollTimeline` IDL](https://drafts.csswg.org/scroll-animations-1/#scrolltimeline) reads as follows:

```
dictionary ScrollTimelineOptions {
  Element? source;
  ScrollAxis axis = "block";
};

[Exposed=Window]
interface ScrollTimeline : AnimationTimeline {
  constructor(optional ScrollTimelineOptions options = {});
};
```

And its [specification text](https://drafts.csswg.org/scroll-animations-1/#dom-scrolltimeline-scrolltimeline) is:

> 1. Let timeline be the new [ScrollTimeline](https://drafts.csswg.org/scroll-animations-1/#scrolltimeline) object.
> 2. Set the [source](https://drafts.csswg.org/scroll-animations-1/#dom-scrolltimeline-source) of timeline to:
> - If the source member of options is present and not null,
>     The source member of options.
> - Otherwise,
>     The [scrollingElement](https://drafts.csswg.org/cssom-view-1/#dom-document-scrollingelement) of the [Document](https://dom.spec.whatwg.org/#document) [associated](https://html.spec.whatwg.org/multipage/window-object.html#concept-document-window) with the [Window](https://html.spec.whatwg.org/multipage/nav-history-apis.html#window) that is the [current global object](https://html.spec.whatwg.org/multipage/webappapis.html#current-global-object).

To me, the IDL and the spec text are contradictory. Since the constructor is specified as taking `optional ScrollTimelineOptions options = {}` then I don't think we can distinguish between `ScrollTimeline` being constructed without an `options` parameter or with `{ source: null }`. But I believe the spec text is actually trying to distinguish between three cases:

1. `{}` where the source would be set to `document.scrollingElement`,
2. `{ source: null }` where the source would not be set,
3. `{ source: document.body }` where the source would be set to `document.body`.

At least this is what is being tested by the WPT test [scroll-animations/scroll-timelines/constructor.html](https://github.com/web-platform-tests/wpt/blob/master/scroll-animations/scroll-timelines/constructor.html).

**Comment by graouts** [comment]
Cc @flackr, @kevers-google and @andruud.

**Comment by cdumez** [comment]
Note that the Web IDL spec says:
> If the type of an argument is a [dictionary type](https://webidl.spec.whatwg.org/#idl-dictionary) or a [union type](https://webidl.spec.whatwg.org/#dfn-union-type) that has a [dictionary type](https://webidl.spec.whatwg.org/#idl-dictionary) as one of its [flattened member types](https://webidl.spec.whatwg.org/#dfn-flattened-union-member-types), and that dictionary type and its ancestors have no [required](https://webidl.spec.whatwg.org/#required-dictionary-member) [members](https://webidl.spec.whatwg.org/#dfn-dictionary-member), and the argument is either the final argument or is followed only by [optional arguments](https://webidl.spec.whatwg.org/#dfn-optional-argument), then the argument must be specified as optional and have a default value provided.

**Comment by graouts** [comment]
To match the WPT test, I believe that:

1. the IDL should be set to `constructor(optional ScrollTimelineOptions options)`,
2. the spec text should be updated to say "If the source member of options is present" and not "If the source member of options is present _and not null_".

That way implementations could distinguish between the three cases outlined above.

**Comment by graouts** [comment]
Upon closer inspection, the IDL constructor definition seems fine because `ScrollTimelineOptions` does _not_ define `source` to default to `null`, so it _is_ possible to distinguish between `source` being explicitly set to `null` and `source` not being explicitly set.

So I think the only required change is to update the spec text as I wrote just above. I'll make a PR to that effect.

**Comment by bramus** [comment]
Fixed by #11357. Closing.
